### PR TITLE
Encoders

### LINKED ISSUE #11: Add ESMEncoder as an option for input
Instead of using the usual encoder, add option of using ESM

### LINKED ISSUE #30: Modularize the encoders
- [ ] rewrite original encoder as GraphEncoder
- [ ] Add PretrainedEncoder
- [ ] other encoders, maybe SequenceEncoder

### LINKED ISSUE #39: Remove the subprocess calls from snakemake testing
`snakemake.snakemake` allows you to call snakemake from a python script (and fast).

### PR DESCRIPTION
**Pull request by ilsenatorov**
closes #30, closes #39, closes #11

### CODE REVIEW

**Review by ilsenatorov — COMMENTED** [review]
Most important changes:

- moved all node and edge encoding to a separate file in `workflow/scripts` 
- removed old redundant pretraining approaches
- moved encoders to a separate folder in rindti, old Encoder is now GraphEncoder
- added esmencoder
- added encoder testing
- snakemake testing is now faster since it's called without `subprocess`

**Comment by Old-Shatterhand** [comment]
1. Please check the `update_config` function again (`rindti/data/datamodules.py:60`). I regenerated the pickle-file with the new snakemake scripts, but there is no `"data"` section in the `"snakemake"` config, i.e. `self.config["snakemake"]["data"]` does not exist, `KeyError` because of `"data"`.
2. `rindti/layers/encoder/pretrained_encoder.py` has no `update_params` function. I guess, it's for loading pretrained weights? I defined it and left it empty so far.
3. In preprocessing, the esm-embeddings are a tensor of size `[1280]`. There is no possibility to call `.size(1)`. Maybe, you changed the esm-generation so that there is `.size(1)` possible, then you can undo my change, it's `workflow/scripts/utils.py:53`

**Comment by Old-Shatterhand** [comment]
I fixed the loss for classification again. It's BCE with logits, which is working for half precision. BCE with logits is the same as a sigmoid layer followed by a BCE loss.
